### PR TITLE
ContactStore

### PR DESCRIPTION
**Pull request by shubhamkmr04**

### CODE REVIEW

**Comment by kaloudis** [comment]
Somewhat on the right track, but the initial loading of contacts should happen when the wallet loads up. We don't want to read from encrypted storage every time we go to contact details, we just want to read from the store

File: stores/ContactStore.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        try {
+            const contactsString = await EncryptedStorage.getItem(
+                'zeus-contacts'
+            );
+            const existingContacts: Contact[] = contactsString
+                ? JSON.parse(contactsString)
+                : [];
+
+            if (isEdit && this.prefillContact && !isNostrContact) {
+                console.log('heree', this.prefillContact);
```
Comment (kaloudis) [inline stores/ContactStore.ts:48]: remove this

**Review by kaloudis — CHANGES_REQUESTED** [review]
change requested

File: stores/ContactStore.ts (hunk 1 above)
Comment (shubhamkmr04) [reply stores/ContactStore.ts]: done

File: views/Settings/Contacts.tsx [hunk 2]
```diff
@@ -463,7 +438,7 @@ export default class Contacts extends React.Component<
                             secondary
                         />
                     )}
-                    {loading ? (
+                    {ContactStore?.loading ? (
```
Comment (kaloudis) [inline views/Settings/Contacts.tsx:441]: since loading is used twice (here and 405), let's destructure it on line 206

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: views/ContactDetails.tsx [hunk 3]
```diff
@@ -672,8 +681,10 @@ export default class ContactDetails extends React.Component<
                                 <Button
                                     onPress={() => {
                                         navigation.goBack();
+                                        ContactStore.setPrefillContact(
```
Comment (kaloudis) [inline views/ContactDetails.tsx:684]: since `setPrefillContact` is used twice (here and 242), let's destructure it at the top of the render function

**Review by kaloudis — CHANGES_REQUESTED** [review]

**Review by kaloudis — APPROVED** [review]

**Pull request merged**